src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiRegularIndex.java

```diff
@@ -91,77 +113,231 @@ public ApiRegularIndex create(
+          };
+
+      if (targetTypeName != ApiTypeName.TEXT && targetTypeName != ApiTypeName.ASCII) {
         // Only text and ascii fields can have the text analysis options specified
-        if (optionsDesc != null) {
-          var anyPresent =
-              optionsDesc.ascii() != null
-                  || optionsDesc.caseSensitive() != null
-                  || optionsDesc.normalize() != null;
-
-          if (anyPresent) {
-            throw SchemaException.Code.UNSUPPORTED_TEXT_ANALYSIS_FOR_DATA_TYPES.get(
-                errVars(
-                    tableSchemaObject,
-                    map -> {
-                      map.put(
-                          "allColumns",
-                          errFmtApiColumnDef(tableSchemaObject.apiTableDef().allColumns()));
-                      map.put("unsupportedColumns", errFmt(apiColumnDef));
-                    }));
-          }
+        var anyPresent =
+            optionsDesc.ascii() != null
+                || optionsDesc.caseSensitive() != null
+                || optionsDesc.normalize() != null;
+
+        if (anyPresent) {
+          throw SchemaException.Code.UNSUPPORTED_TEXT_ANALYSIS_FOR_DATA_TYPES.get(
+              errVars(
+                  tableSchemaObject,
+                  map -> {
+                    map.put(
+                        "allColumns",
+                        errFmtApiColumnDef(tableSchemaObject.apiTableDef().allColumns()));
+                    map.put("unsupportedColumns", errFmt(apiColumnDef));
+                  }));
         }
-        // nothing to update in the cqlOptions for these indexes
-      } else {
-        // text and ascii fields can have the text analysis options specified
-        var ascii =
-            ApiPropertyUtils.getOrDefault(
-                optionsDesc,
-                RegularIndexDefinitionDesc.RegularIndexDescOptions::ascii,
-                TableDescDefaults.RegularIndexDescDefaults.ASCII);
-        put(indexOptions, CQLOptions.ASCII, ascii);
-
-        var case_sensitive =
-            ApiPropertyUtils.getOrDefault(
-                optionsDesc,
-                RegularIndexDefinitionDesc.RegularIndexDescOptions::caseSensitive,
-                TableDescDefaults.RegularIndexDescDefaults.CASE_SENSITIVE);
-        put(indexOptions, CQLOptions.CASE_SENSITIVE, case_sensitive);
-
-        var normalize =
-            ApiPropertyUtils.getOrDefault(
-                optionsDesc,
-                RegularIndexDefinitionDesc.RegularIndexDescOptions::normalize,
-                TableDescDefaults.RegularIndexDescDefaults.NORMALIZE);
-        put(indexOptions, CQLOptions.NORMALIZE, normalize);
       }
+    }
+
+    /**
+     * Method to populate the index options map.
+     *
+     * <p>Note, this method should be called after the options are being validated.
+     */
+    private void populateIndexOptions(
+        RegularIndexDefinitionDesc.RegularIndexDescOptions optionsDesc,
+        Map<String, String> indexOptions) {
```

Comment:
nit: there is no need for indexOptions map to be passed in, it can be the return